ibis/expr/api.py

```diff
@@ -678,12 +678,21 @@ def cast(arg, target_type):
     if op.to.equals(arg.type()):
         # noop case if passed type is the same
         return arg
-    else:
-        result = op.to_expr()
-        if not arg.has_name():
-            return result
-        expr_name = 'cast({}, {})'.format(arg.get_name(), op.to)
-        return result.name(expr_name)
+
+    if (
+        target_type.lower() in ('geometry', 'geography')
+        and (
+            arg.type().geotype == target_type
+            or (arg.type().geotype is None and target_type == 'geometry')
```

Comment:
Can you explain the behavior here? I'm not sure I follow purpose of this additional logic. In particular, what is the rationale for not performing a cast in these conditions?

It looks like you're trying to enable generic "cast to geometry or geography" functionality, and trying to avoid generating code for cast that would essentially be a no-op. Is there any reason to avoid generating this code?

I want to make sure we're not baking in assumptions that are tied to the OmniSci backend in the generic expression API, so if you could walk me through the reasoning here that would be helpful.

Reply:
It might help to add a unit test _just_ for this functionality.

Reply:
well I will try to explain me.

In this PR the cast operation for geospatial works from point, linestring, polygon or multipolygon to geometry or geography.

point, linestring, polygon and multipolygon have a attribute called geotype that specify it is a geometry or geography (also it could be None, that is treated as geometry)

so cast from geometry to geometry is the same to cast int to int ... so I am follow the same idea of `if op.to.equals(arg.type()): ...`

currently omniscidb doesn't support cast to geometry ... but I am treating that just inside the mapd backend.

Reply:
Ok, makes a bit more sense now. Can you perform the comparison with `op.to` instead of looking at `target_type`? `op.to` is an instance of `DataType` as opposed to `target_type` which can be either a string _or_ a `DataType`.